### PR TITLE
Wireframes

### PR DESCRIPTION
**Pull request by PMcS3**
@jfarmer Just realized you couldn't see these - I chose to replicate the first Wireframe using Bootstrap

### CODE REVIEW

File: a4/a4.html [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title> ShopQuick </title>
+
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
```
Comment (jfarmer) [inline a4/a4.html:9]: Careful, you're using an old version of Bootstrap. The current version is 4.4. The 3.4 page displays a message saying there's a newer version.

This isn't a problem _per se_, but it does mean you have to be careful when reading documentation. Examples for 4.4 likely won't work.

File: a4/a4.html [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+    <style>
+      article {
+        margin: 250px;
```
Comment (jfarmer) [inline a4/a4.html:14]: Let Botostrap handle this sort of thing.

File: a4/a4.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        <div class="col-sm-4">
+          <h3>For Fun</h3>
+        </div>
+      </div>
+    </div>
+
+    <div class="container" id="space">
+      <div class="row">
+        <div class="col-sm-4">
+          <h3></h3>
```
Comment (jfarmer) [inline a4/a4.html:50]: The browser isn't going to like an empty tag here.

Regardless, see https://getbootstrap.com/docs/3.4/css/#grid-example-basic for how to offset columns. You don't need to define empty columns.

File: a4/a4.html [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        </div>
+        <div class="col-sm-4">
+          <h3>Newly added items!</h3>
+        </div>
+        <div class="col-sm-4">
```
Comment (jfarmer) [inline a4/a4.html:55]: If you only specify the first 8 of the 12 columns in a row, the remaining 4 will be left blank (per the Bootstrap docs). No need to include a blank column here.

File: a4/a4.html [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          <h3></h3>
+        </div>
+      </div>
+    </div>
+
+    <div class="container">
```
Comment (jfarmer) [inline a4/a4.html:61]: Look around the Bootstrap site for examples of what you want to do (or similar).

I found this after clicking through the components one after another: https://getbootstrap.com/docs/3.3/components/#thumbnails

Bootstrap 4 elevated this pattern to a named thing in the system: https://getbootstrap.com/docs/4.3/components/card/

**Review by jfarmer — COMMENTED** [review]
1. Use the validator: https://validator.w3.org/nu/?doc=https%3A%2F%2Fexcellent-circle.surge.sh%2Fa4.html
2. Spend more time poking around the components your CSS framework provides. For Bootstrap 3, everything here: <https://getbootstrap.com/docs/3.4/components/>

   Look for examples that are similar to the effect you want to achieve.

File: a4/a4.html (hunk 2 above)
Comment (PMcS3) [reply a4/a4.html:14]: For top/bottom margin, I continued like this. I used a Bootstrap `mx-auto` class to center my button group on the page. Should use the Bootstrap classes to dictate all of my margins?

**Comment by PMcS3** [comment]
@jfarmer I used Bootstrap 4 this time around, as that was the documentation I was reading. It was way more intuitive.

**Pull request closed without merging**